main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -1156,6 +1156,28 @@ object GenExpression {
       // Emit code for the invocation of entails.
       visitor.visitMethodInsn(INVOKESTATIC, JvmName.Runtime.Fixpoint.Solver.toInternalName, "entails", "(Lflix/runtime/fixpoint/ConstraintSystem;Lflix/runtime/fixpoint/ConstraintSystem;)Z", false);
 
+    case Expression.FixpointFold(pred, init, f, constraints, tpe, loc) =>
+      // Add source line numbers for debugging.
+      addSourceLine(visitor, loc)
+
+      // Instantiate the predicate symbol.
+      newPredSym(pred, visitor)(root, flix, currentClass, lenv0, entryPoint)
+
+      // Compile the init argument
+      compileExpression(init, visitor, currentClass, lenv0, entryPoint)
+
+      // Compile the folded function
+      compileExpression(f, visitor, currentClass, lenv0, entryPoint)
+
+      // Compile the constraint system.
+      compileExpression(constraints, visitor, currentClass, lenv0, entryPoint)
+
+      // Emit the fold
+      ??? // TODO
+      // Option 1: Call a method of ConstraintSystem. But that method has to be able to call function f, and to translate facts into tuples
+      // Option 2: Directly emit the code that loops over the facts, translates them into tuples, and calls function f
```

Comment:
I think option 2 is reasonable. You can get the facts from the constraint system as an array (possibly by calling a helper method to collect the facts of the right symbol) and then you can write code here to iteratively call the fold function. 

One issue, for performance, is that you should not have a loop with a non-empty stack height. Thus you would need local variables to hold the two last expressions, but then you need to introduce local variables for `f` and `constraints`.